schemas/tests.yaml

```diff
@@ -0,0 +1,158 @@
+---
+
+#
+# JSON Schema definition for tmt `Tests`
+# https://tmt.readthedocs.io/en/stable/spec/tests.html
+#
+
+definitions:
+
+  beakerlib_library:
+    type: array
+    minItems: 1
+
+    items:
+      oneOf:
+        # library(repo/lib)
+        - type: string
+          pattern: "^library([a-zA-Z0-9]+/[a-zA-Z0-9]+)"
+
+        # url: https://github.com/beakerlib/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - url
+
+        # path: /tmp/library/openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+
+            path:
+              type: string
+              pattern: "^/.+"
+
+          required:
+            - name
+            - path
+
+        # url: https://github.com/redhat-qe-security/certgen
+        # ref: devel
+        # nick: openssl
+        # name: /certgen
+        - type: object
+
+          properties:
+            name:
+              type: string
+            nick:
+              type: string
+            ref:
+              type: string
+            url:
+              type: string
+              pattern: "^https?://"
+
+          required:
+            - name
+            - nick
+            - ref
+            - url
+
+  string_or_array:
+    oneOf:
+      - type: string
+
+      - type: array
+        minItems: 1
+
+$id: tests.yaml
+
+$schema: https://json-schema.org/draft-07/schema
+
+type: object
+
+additionalProperties: false
+
+properties:
+
+  # TODO: adjust
+
+  component:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  contact:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  description:
+    type: string
+
+  duration:
+    type: string
+    pattern: "^[0-9]+[smhd]$"
+
+  enabled:
+    type: string
+    pattern: "^(true|false)$"
+
+  environment:
+    type: object
+
+    minProperties: 1
+
+    additionalProperties:
+      type: string
+
+  framework:
+    type: string
+    pattern: "^(beakerlib|shell)$"
+
+  # TODO: link
+
+  manual:
+    type: string
+    pattern: "^(true|false)$"
+
+  order:
+    type: integer
+    multipleOf: 1
+
+  path:
+    type: string
+    pattern: "^/.+"
+
+  recommend:
+    $ref: "tests.yaml#/definitions/string_or_array"
+
+  require:
+    oneOf:
+      - type: string
+      - $ref: "tests.yaml#/definitions/beakerlib_library"
```

Comment:
I think this does not cover the "list of strings/beaker libraries/both" variant, only values that are not arrays. What you have here is right, but there's the third `oneOf` which is a `array` where items are `oneOf` string or beaker lib.

Reply:
thx, will check

Reply:
added array, and I belive it can be anyOf?

Reply:
I mentioned it in another comment, just adding it here for the sake of completeness: `require: []` is out there already, used by tmt users. I used it myself in several tests, part of my motivation was to make sure all my tests specify `require`, and with `require: []` where no requirements are needed, it's then simpler to search for files *without* `require` rather than to decide whether `require` is missing or not needed at all.

Reply:
Removed `minItems`